yt/geometry/coordinates/cartesian_coordinates.py

```diff
@@ -36,6 +36,71 @@
 from yt.utilities.nodal_data_utils import get_nodal_data
 from yt.units.yt_array import uconcatenate
 
+
+def _find_object_bbox(data_source):
```

Comment:
I think it would be cleaner to make this a method of `YTSelectionContainer3D` that subclasses implement. That makes it clearer for implementors of future data objects that they'd need to touch this code.

Reply:
I agree!

Reply:
One other thing though--these rules only really work for cartesian datasets. I guess there's nothing we can do about that now?

Reply:
Mayve check the geometry of the dataset and raise a NotImplementedError in the base class implementation? Right now we’re only using this for SPH data and non-cartesian SPH data is not something we’ve thought about at all so in practice that error won’t be hit unless someone tries to repurpose this for ither things in the future, at which point we can come back to this.